data/mil2525d/utilities/military-features-utilities/template-gdb-toolbox/scripts/CreateTemplateGDB/AddFieldsFromSchema.py

```diff
@@ -69,38 +131,45 @@ def addFieldsFromSchema(schemasFolder, featureClass, schema):
+                        # Add a line as a new field if this is the first occurence of that "field"
                         
-                        if fieldType == "TEXT":
-                            length = int(line[2])
+                        if line[0] != fieldName:
+                            addAField(featureClass, fieldType, line)
+                            fieldName = line[0]
                             
-                            
-                            arcpy.AddField_management(featureClass, \
-                                                      field_name=line[0], \
-                                                      field_type=fieldType, \
-                                                      field_length=length, \
-                                                      field_alias=line[3], \
-                                                      field_is_nullable=line[4], \
-                                                      field_domain=line[5])
-                        else:
-                            arcpy.AddField_management(featureClass, \
-                                                      field_name=line[0], \
-                                                      field_type=fieldType, \
-                                                      field_alias=line[3], \
-                                                      field_is_nullable=line[4], \
-                                                      field_domain=line[5])
+                        # Set the domain for the field, including subtype code if necessary
                         
+                        if bool(line[5]):
+                            if bool(line[8]):
+                                arcpy.AssignDomainToField_management(featureClass, line[0], line[5], line[8] + ": " + subTypes[line[8]])
+                            else:
+                                arcpy.AssignDomainToField_management(featureClass, line[0], line[5])
+                            
                         # Set the default value for the field
                         
                         if bool(line[6]):
                             defaultValue = line[6]
                             castedDefault = castValue(fieldType, defaultValue)
-                            arcpy.AssignDefaultToField_management(featureClass, line[0], castedDefault)
+                            
+                            if bool(line[8]):
+                                arcpy.AssignDefaultToField_management(featureClass, line[0], castedDefault, line[8] + ": " + subTypes[line[8]])
+                            else:
+                                arcpy.AssignDefaultToField_management(featureClass, line[0], castedDefault)
+                                
+                        # Check to see if this field sets a subtype and if it does, go create the subtypes
+                        
+                        if line[7] == "True":
```

Comment:
Rather than "True"/"False" for this column i.e. "line[7]"(= IsSubTypeDefined?) (which seems like unnecessary data bloat and doesn't make it clear where this subtype is even coming from), couldn't you just put the subtype file/entry here instead: "Subtype_XXXX" - we wouldn't have all those superfluous "False" entries & it would be clearer what the relationship was between this entry and that file.